### PR TITLE
CE-93: nitpickery

### PR DESCRIPTION
**Pull request by ialarmedalien**
- new `checkAgainstRegex` method to allow regex matching and optional string replacement
- variable renaming for greater accuracy and consistency

### CODE REVIEW

**Comment by MrCreosote** [comment]
Looks like there's some uncovered lines in the new code

File: src/us/kbase/workspace/database/provenance/Common.java [hunk 1]
```diff
@@ -55,27 +55,62 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
...
 	 *
-	 * @param putativePid the string to check.
-	 * @param name        the name of the string to use in any error messages.
-	 * @param optional    whether or not the field is optional. If true, null is a valid value for the PID.
-	 * @return the trimmed PID.
+	 * @param stringToCheck the string to check.
+	 * @param pattern       the pattern to validate against.
+	 * @param replace       if non-null, the pattern to use for the replaceAll operation.
+	 * @param name          the name of the string to use in any error messages.
+	 * @param optional      whether or not the field is optional. If true, null is a valid value for the field.
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/Common.java:67]: Nit: null or whitespace

File: src/us/kbase/workspace/database/provenance/Common.java [hunk 2]
```diff
@@ -55,27 +55,62 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
...
+	}
+
+	/**
+	 * Trims leading and trailing whitespace, and then checks that a string is either null, or
+	 * has at least one non-whitespace character and conforms to the specified regular expression.
+	 *
+	 * @param stringToCheck the string to check.
+	 * @param pattern       the pattern to validate against.
+	 * @param name          the name of the string to use in any error messages.
+	 * @param optional      whether or not the field is optional. If true, null is a valid value for the field.
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/Common.java:95]: nit whitespace

File: src/us/kbase/workspace/database/provenance/Common.java [hunk 3]
```diff
@@ -55,27 +55,62 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
...
+		return checkAgainstRegex(stringToCheck, pattern, null, name, optional);
+	}
+
+	/**
+	 * Trims leading and trailing whitespace, and then checks that a putative PID is either null,
+	 * or has at least one non-whitespace character and conforms to the specified regular expression.
+	 *
+	 * @param putativePid the putative PID string.
+	 * @param name        the name of the string to use in any error messages.
+	 * @param optional    whether or not the field is optional. If true, null is a valid value for the field.
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/Common.java:108]: nit whitespace

**Review by MrCreosote — CHANGES_REQUESTED** [review]
The `checkAgainstRegex` methods need their own tests but otherwise LGTM

File: src/us/kbase/workspace/database/provenance/Common.java (hunk 1 above)
Comment (ialarmedalien) [reply src/us/kbase/workspace/database/provenance/Common.java:67]: > Trims leading and trailing whitespace and then checks that a string is either null, or has at least ...

Whitespace isn't a valid value; null is.

**Comment by ialarmedalien** [comment]
I've got tests for `checkAgainstRegex` incorporated into the next PR.

File: src/us/kbase/workspace/database/provenance/Common.java (hunk 1 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/database/provenance/Common.java:67]: Unless I'm completely misreading something here, if optional is true both `null` and `\t   \n` will result in this function returning `null`... right?

Comment (MrCreosote) [reply src/us/kbase/workspace/database/provenance/Common.java:67]: Maybe the confusion is over value vs. input, I was thinking input

**Comment by MrCreosote** [comment]
Is there a reason to not have them in this PR? I try to always have tests against any new code in a PR

File: src/us/kbase/workspace/database/provenance/Common.java (hunk 1 above)
Comment (ialarmedalien) [reply src/us/kbase/workspace/database/provenance/Common.java:67]: Yup, that's it. How about this wording for the param:

```
	 * @param optional      whether or not the field is optional. If false, null,
         *                      empty, or whitespace-only input strings will throw an error.
```

Comment (MrCreosote) [reply src/us/kbase/workspace/database/provenance/Common.java:67]: :+1:

**Comment by ialarmedalien** [comment]
The appropriate tests will be in the CE-94_resolvable_pid PR (#636), which I'll move out of draft mode once it's rebased and updated.

**Comment by MrCreosote** [comment]
In order to get the tests to pass, `next` will need to be changed to `ci` here:
https://github.com/kbase/workspace_deluxe/blob/develop/src/us/kbase/common/test/service/ServiceCheckerTest.java#L20

Probably a better option in any case given the evolution of next and ci

**Comment by MrCreosote** [comment]
also the testPerl test can be dropped given there aren't any Perl SDK core services left

**Review by MrCreosote — CHANGES_REQUESTED** [review]
This looks good other than the fix to get the tests to pass

**Comment by MrCreosote** [comment]
Codecov clearly has the report for the commit but doesn't seem to be able to update the comment:
https://app.codecov.io/github/kbase/workspace_deluxe/commit/7bfd1e7375d8a7c3f77dcbc6d4e77e4f39fb3305

**Comment by MrCreosote** [comment]
Or update the pull request either https://app.codecov.io/github/kbase/workspace_deluxe/pull/639
